### PR TITLE
Abhi132004 patch 3

### PR DESCRIPTION
**Pull request by Abhi132004**
To provide better error messages when the grpc command crashes, I have also considered improving the error handling within the ''generateBalFile'' method and other relevant methods in the same Java file. Look for any potential exceptions that may occur during protocol buffer processing and handle them appropriately by providing informative error messages.

Isse: https://github.com/ballerina-platform/ballerina-standard-library/issues/4656

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec

### CODE REVIEW

File: protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java [hunk 1]
```diff
@@ -125,46 +125,38 @@ private static void exportResource(String resourceName, ClassLoader classLoader)
...
             if (pathExtension.isEmpty() || !PROTO_EXTENSION.equalsIgnoreCase(pathExtension.get())) {
-                String errorMessage = "Invalid proto file path. Please input valid proto file location.";
-                outStream.println(errorMessage);
-                return;
+                throw new RuntimeException("Input directory does not contain any proto files. " +
+            "Please input a valid proto files directory.");
             }
             generateBalFile(protoPath);
         }
+    } catch (Exception e) {
```
Comment (dilanSachi) [inline protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java:156]: Even if we do a catch here, the error is logged elsewhere and also printed to the stdout. The exception is not thrown to here. Hence, this catch will not be able to catch it. We need to remove the log with the stacktrace and display a reason for the failure if possible.

Comment (Abhi132004) [reply protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java:156]: catch (Exception e) {
        // Handle the exception and print a user-friendly error message
        handleException(e);
    }
}

private void handleException(Exception e) {
    // Log the exception or print a user-friendly error message
    // If you want to suppress the stack trace, you can just print a custom error message here
    outStream.println("Error: " + e.getMessage());
}

In this modified code, the catch block in the execute method calls a handleException method, which you can customize to log the exception, print an error message, or suppress the stack trace as needed. This allows you to have control over how the exception is handled without the default stack trace logging.

**Comment by Abhi132004** [comment]
> ## [Codecov](https://app.codecov.io/gh/ballerina-platform/protoc-tools/pull/45?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ballerina-platform) Report
> Attention: `11 lines` in your changes are missing coverage. Please review.
> 
> > Comparison is base [(`ffbf33d`)](https://app.codecov.io/gh/ballerina-platform/protoc-tools/commit/ffbf33d374ae45451b3f5f815c3932fc5bd64610?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ballerina-platform) 86.42% compared to head [(`771f52f`)](https://app.codecov.io/gh/ballerina-platform/protoc-tools/pull/45?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ballerina-platform) 86.53%.
> 
> Additional details and impacted files
> [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/ballerina-platform/protoc-tools/pull/45?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ballerina-platform). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ballerina-platform).

File: protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java (hunk 1 above)
Comment (Abhi132004) [reply protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java:156]: @dilanSachi

**Comment by Abhi132004** [comment]
@dilanSachi @daneshk if all the things are okay can I fill the google form?

File: protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java (hunk 1 above)
Comment (dilanSachi) [reply protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java:156]: @Abhi132004. It seems you have not understood what I meant in my comment. Let me try and explaing a bit more. The issue is related to the `generateBalFile` method. In this method, when there are errors in the descriptor generator phase, they are caught as a `CodeGeneratorException` in here [1]. However, we have only printed a generic error and the error message. But this prints stack traces as well. What we need to do is to check if we can add any error information to this printed error and try and stop the stack trace printing in the cmd.

[1] https://github.com/ballerina-platform/protoc-tools/blob/6db60af220d182506a59f49d8d527ccaa4953b4d/protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java#L265

File: protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java [hunk 2]
```diff
@@ -125,47 +125,74 @@ private static void exportResource(String resourceName, ClassLoader classLoader)
...
+        // Handle other exceptions if needed
+        // You can choose to log, print, or handle them differently
+        outStream.println("Unexpected Error: " + e.getMessage());
     }
+}
+private void generateBalFile(String protoFile) {
```
Comment (dilanSachi) [inline protoc-cli/src/main/java/io/ballerina/protoc/protobuf/cmd/GrpcCmd.java]: why have u created a new method?
There's already a method with implementations. And also there are compile errors

**Pull request closed without merging**